pallets/mint_with_fees/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+pub use frame_support::{
+        dispatch::DispatchResult, pallet_prelude::*,
+        traits::Currency,
+        sp_runtime::traits::{CheckedConversion, CheckedDiv}
+    };
+pub use frame_system::{
+        pallet_prelude::*,
+        ensure_root,
+    };
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: Currency<Self::AccountId>;
+	}
+
+    // type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    // type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// #[pallet::storage]
+	// #[pallet::getter(fn something)]
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// the value have been minted on the target account
+		/// [target_account, value]
+		ValueMinted(T::AccountId, u128),
+		/// the fees have been minted on the nsm account
+		/// [nsp_account, value]
+		FeesMinted(T::AccountId, u128),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+        /// Overflow when adding the value to the account
+		Overflow,
+        /// Amount is 0
+        ZeroAmount,
+	}
+
+    #[pallet::storage]
+    #[pallet::getter(fn percent_for_nsp)]
+    pub type PercentForNSM<T: Config> = StorageValue<_, u8>;
```

Comment:
Use 
`    pub type PercentForNSM<T: Config> = StorageValue<_, u8, ValueQuery>;`
It make the getter return the value, and default value if not set. Instead of an option.
You can also specify your own default value.
https://docs.substrate.io/v3/runtime/storage/#default-values